mesonbuild/environment.py

```diff
@@ -1757,6 +1759,30 @@ def detect_cs_compiler(self, for_machine):
 
         self._handle_exceptions(popen_exceptions, compilers)
 
+    def detect_cython_compiler(self, for_machine: MachineChoice) -> CythonCompiler:
+        """Search for a cython compiler."""
+        compilers = self.lookup_binary_entry(for_machine, 'cython')
```

Comment:
I wonder whether this should better use the `cython` or the `cythonize` frontend?

Reply:
Pretty sure it must be `cython`. Parallel builds, caching, etc. should come from Meson not Cython, and we want Meson to invoke the C/C++ compiler rather than have Cython use setuptools under the hood.

Reply:
I'd prefer to build on `cython` not `cythonize`, we can easily generate a .pyx -> .c rule and a second .c -> .o rule ourselves. Vala already works this way.